lighthouse/src/main.rs

```diff
@@ -514,6 +544,15 @@ fn run<E: EthSpec>(
                 serde_json::to_writer(&mut file, &config)
                     .map_err(|e| format!("Error serializing config: {:?}", e))?;
             };
+            if let Some(dump_path) =
+                clap_utils::parse_optional::<PathBuf>(matches, "dump-chain-config")?
+            {
+                let chain_config = Config::from_chain_spec::<E>(&context.eth2_config.spec);
+                let mut file = File::create(dump_path)
+                    .map_err(|e| format!("Failed to create dumped chain config: {:?}", e))?;
+                serde_yaml::to_writer(&mut file, &chain_config)
+                    .map_err(|e| format!("Error serializing chain config: {:?}", e))?;
+            };
```

Comment:
Not a huge deal, but there's starting to be a bit of duplication between the BN and VC with some of the code that runs here.

It _might_ be worth spinning it out into `dump_config` and `dump_chain_config` functions which can be reused between both the BN and VC? If it lived in `clap_utils` (or somewhere else appropriate) they might also be able to be used for the `boot_node`'s `dump-chain-config` flag. (Although you may run into type issues)

If you wanted to get extra fancy, it would be really cool if it was just 1 function which was generic over different `Config`'s but that's probably more work than it's worth.

Reply:
A function that's generic over some `config: T` where `T: serde::Serialize` would be pretty simple and do the trick, I think.

Reply:
Done! Let me know what you guys think..